credsweeper/utils/util.py

```diff
@@ -218,3 +219,41 @@ def preprocess_file_diff(changes: List[DiffDict]) -> List[DiffRowData]:
                 rows_data.append(DiffRowData(DiffRowType.DELETED_ACCOMPANY, change["old"], change["line"]))
 
         return rows_data
+
+    @staticmethod
+    def is_zip(data: bytes) -> bool:
+        """According https://en.wikipedia.org/wiki/List_of_file_signatures"""
+        if isinstance(data, bytes) and 3 < len(data):
+            # PK
+            if 0x50 == data[0] and 0x4B == data[1]:
+                if 0x03 == data[2] and 0x04 == data[3]:
+                    return True
+                # empty archive - no sense to scan
+                elif 0x05 == data[2] and 0x06 == data[3]:
+                    return True
```

Comment:
No need to scan: we may safely skip it from scanning procedure by returning False
```suggestion
                    return False
```

Reply:
It skips scanning the file with TextContentProvider.